src/pdf/ShoppingList/dataPreparation.ts

```diff
@@ -0,0 +1,236 @@
+    const address: string[] = [];
+    [address_1, address_2, address_county, address_postcode, address_town].forEach((value) => {
+        if (value !== "") {
+            address.push(value);
+        }
+    });
+
+    return {
+        name: clientData.full_name,
+        contact: clientData.phone_number,
+        address: address.join("\n"),
+        extraInformation: "",
+    };
+};
+
+const getChild = (child: Person): string => {
+    return `${child.age} ${child.gender[0].toUpperCase()}`;
+};
+
+const displayList = (data: string[]): string => {
+    return data.length === 0 ? "None" : data.join(", ");
+};
+
+const prepareHouseholdSummary = (familyData: Schema["families"][]): HouseholdSummary => {
+    const children = familyData.filter((member) => member.age !== null);
+
+    const householdSize = familyData.length;
+    const numberBabies = familyData.filter((member) => member.age === 0).length;
+    const numberFemales = familyData.filter((member) => member.gender === "female").length;
+    const numberMales = familyData.filter((member) => member.gender === "male").length;
+
+    return {
+        householdSize: `${householdSize} (${householdSize - children.length} Adults ${
+            children.length
+        } Children)`,
+        genderBreakdown: `${numberFemales} Female ${numberMales} Male ${
+            householdSize - numberFemales - numberMales
+        } Other`,
+        numberOfBabies: numberBabies.toString(),
+        ageAndGenderOfChildren: displayList(children.map(getChild)),
+    };
+};
+
+const prepareRequirementSummary = (clientData: Schema["clients"]): RequirementSummary => {
+    let babyProduct: string;
+    if (clientData.baby_food) {
+        babyProduct = "Yes";
+    } else {
+        babyProduct = clientData.baby_food === null ? "Don't Know" : "No";
+    }
+
+    return {
+        feminineProductsRequired: displayList(clientData.feminine_products),
+        babyProductsRequired: babyProduct,
+        petFoodRequired: displayList(clientData.pet_food),
+        dietaryRequirements: displayList(clientData.dietary_requirements),
+        otherItems: displayList(clientData.other_items),
+    };
+};
+
+const getQuantityAndNotes = (
+    row: Schema["lists"],
+    size: number
+): Pick<Item, "quantity" | "notes"> => {
+    if (size >= 10) {
+        return { quantity: row["10_quantity"] ?? "", notes: row["10_notes"] ?? "" };
+    }
+    return {
+        quantity: row[`${size}_quantity` as keyof Schema["lists"]] ?? "",
+        notes: row[`${size}_notes` as keyof Schema["lists"]] ?? "",
+    };
+};
+
+const prepareItemsList = async (householdSize: number): Promise<Item[]> => {
+    const listData = await fetchLists();
+    return listData.map((row): Item => {
+        return {
+            description: row.item_name,
+            ...getQuantityAndNotes(row, householdSize),
+        };
+    });
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const getEndNotes = (_parcelID: string): string => {
```

Comment:
Should we have this as a const string variable instead? Feels a little weird to have an incomplete function on main. You could add a TODO here I think.

Reply:
I think we should keep it as a function and I don't think this is incomplete. I think when we have a way of getting the end notes, we'll just need to change the implementation of this function instead of change the main code (which is in `prepareData`. It's just that right now the "implementation" of this function is to always return a constant.